cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -1334,23 +1334,26 @@ data ShelleyWitnessSigningKey =
      | WitnessGenesisDelegateExtendedKey
                                  (SigningKey GenesisDelegateExtendedKey)
      | WitnessGenesisUTxOKey     (SigningKey GenesisUTxOKey)
+     | WitnessMultiSig           MultiSigScript
 
 
 makeShelleyKeyWitness :: TxBody Shelley
-                      -> ShelleyWitnessSigningKey
+                      -> ShelleyWitnessKeyOrScript
                       -> Witness Shelley
```

Comment:
I'm not sure that we should introduce this `ShelleyWitnessKeyOrScript` type. I'd prefer that we have a separate function for `makeShelleyScriptWitness`.